app/hooks/useBitcoinWallet.tsx

```diff
@@ -22,28 +22,42 @@ export const BitcoinWalletProvider: React.FC<BitcoinWalletProviderProps> = ({
   settings,
   children
 }) => {
-  const [wallet, setWallet] = useState({});
+  const [wallet, setWallet] = useState(null);
   const [loading, setLoading] = useState(false);
   const [loaded, setLoaded] = useState(false);
 
   useEffect(() => {
-    async function initializeBitcoinWallet() {
+    const initializeBitcoinWallet = async () => {
       setLoading(true);
 
-      // TODO: upgrade SDK to 0.17.0
-      // https://github.com/comit-network/comit-js-sdk/blob/dev/src/wallet/bitcoin.ts#L71-L79
-      // const w = await InMemoryBitcoinWallet.newInstance(
-      //   'regtest',
-      //   settings.get('BITCOIN_P2P_URI'),
-      //   settings.get('BITCOIN_HD_KEY')
-      // );
-      // await new Promise(resolve => setTimeout(resolve, 1000)); // bitcoin wallet workaround
+      const url = settings.get('BITCOIN_HTTP_URI');
+      const username = settings.get('BITCOIN_USERNAME');
+      const password = settings.get('BITCOIN_PASSWORD');
+      const walletDescriptor = settings.get('BITCOIN_WALLET');
+      const walletName = 'takerui';
 
-      // setWallet(w);
+      const params = {
+        url,
+        username,
+        password,
+        walletDescriptor,
+        walletName,
+        rescan: true,
+        refreshIntervalMs: 10
+      };
 
+      let w;
+      try {
+        w = await BitcoindWallet.newInstance(params);
+      } catch (e) {
+        console.log(e);
```

Comment:
What happens if this fails and `w` is not initialized?

Reply:
Components that use `w` will encounter an error. We can handle this more declaratively with Error boundaries https://reactjs.org/docs/error-boundaries.html but this can be a future enhancement

Reply:
Created follow up https://github.com/comit-network/taker-ui/issues/22